infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/LogReplicationSourceManager.java

```diff
@@ -247,17 +226,75 @@ public LogReplicationEntry receive(LogReplicationEntry message) {
         // Process ACKs from Application, for both, log entry and snapshot sync.
         if(message.getMetadata().getMessageMetadataType() == MessageType.LOG_ENTRY_REPLICATED) {
             log.debug("Log entry sync ACK received on timestamp {}", message.getMetadata().getTimestamp());
+            lastAckedTimestamp.set(message.getMetadata().getTimestamp());
             logReplicationFSM.input(new LogReplicationEvent(LogReplicationEventType.LOG_ENTRY_SYNC_REPLICATED,
                 new LogReplicationEventMetadata(message.getMetadata().getSyncRequestId(), message.getMetadata().getTimestamp())));
         } else if (message.getMetadata().getMessageMetadataType() == MessageType.SNAPSHOT_REPLICATED) {
             log.debug("Snapshot sync ACK received on base timestamp {}", message.getMetadata().getSnapshotTimestamp());
+            lastAckedTimestamp.set(message.getMetadata().getTimestamp());
```

Comment:
For snapshot messages, should it be msgSeqNum? All snapshot messages have the same timestamp.